src/components/shared/IdleSearchField.js

```diff
@@ -86,6 +87,25 @@ class IdleSearchField extends PureComponent {
     }
   }
 
+  _onFormSubmit(e: SyntheticEvent & { currentTarget: HTMLElement }) {
+    e.preventDefault();
+
+    // 1. Notify the current value
+    clearTimeout(this._timeout);
+    this._timeout = 0;
+    this._notifyIfChanged(this.state.value);
+
+    // 2. Notify the user wants to persist this value
+    const { onSubmit } = this.props;
+    if (onSubmit) {
+      onSubmit();
+    }
+
+    // 3. Update our local state
+    this.setState({ value: '' });
+    this._notifyIfChanged('');
```

Comment:
I'm not entirely sure this last step is necessary now (I did various tries while coding). Please tell me if you'd like me to try removing it and test.